lib/covid19_orientation/data/store.ex

```diff
@@ -1,68 +1,89 @@
 defmodule Covid19Orientation.Data.Store do
   @moduledoc """
   Store data in PostgreSQL.
+  """
 
-  Test table:
+  use GenServer
 
-  ```sql
-  DROP TABLE IF EXISTS journal;
-  CREATE TABLE journal (
-      date TIMESTAMP WITHOUT TIME ZONE NOT NULL,
-      uuid VARCHAR NOT NULL,
-      data JSONB NOT NULL
-  );
-  ```
-  """
+  alias Covid19Orientation.Data.{Journal, Repo}
+  alias Ecto.Adapters.SQL
+  require Logger
 
-  use Agent
+  @tick_interval 1000
+  @chunk_size 100
 
-  def start_link(_initial) do
-    Agent.start_link(
-      fn ->
-        settings = Application.fetch_env!(:covid19_orientation, __MODULE__)[:conn_opts]
-        {:ok, pid} = Postgrex.start_link(settings)
-        {:pid, pid}
-      end,
-      name: __MODULE__
-    )
+  def write({date, uuid}, data = %{}) do
+    GenServer.cast(__MODULE__, {:write, %{date: date, uuid: uuid, data: data}})
+    data
   end
 
-  def write({date, uuid}, data = %{}) do
-    Agent.cast(__MODULE__, fn state ->
-      insert(state, date, uuid, data)
-      state
-    end)
+  def read({date, uuid}) do
+    select(date, uuid)
+  end
 
-    data
+  def tick_interval, do: @tick_interval
+
+  #
+  # GenServer Part
+  #
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [to_write: []], name: __MODULE__)
   end
 
-  @spec read(tuple) :: {:ok, map} | {:error, any}
+  @impl true
+  def init(_) do
+    tick()
+    {:ok, [to_write: []]}
+  end
 
-  def read({date, uuid}) do
-    Agent.get(__MODULE__, &select(&1, date, uuid))
+  @impl true
+  def handle_cast({:write, element}, [to_write: to_write]) when length(to_write) < @chunk_size do
+    new_state = [to_write: [element | to_write]]
+
+    {:noreply, new_state}
   end
 
-  defp insert({:pid, pid}, date, uuid, data) do
-    Postgrex.query!(pid, "INSERT INTO journal (date, uuid, data) VALUES($1, $2, $3)", [
-      date,
-      uuid,
-      data
-    ])
+  @impl true
+  def handle_cast({:write, element}, [to_write: to_write]) do
+    insert_many([element | to_write])
```

Comment:
If this doesn't work, there's no way to keep those records somehow in state or DTS ? Does it make sense ? If a live test shows drops of <5% I'm happy though.

Reply:
This is just logging, I think we can just let it fail